pkg/avancement/service_manager.go

```diff
@@ -134,13 +135,45 @@ func (s *ServiceManager) Promote(serviceName, fromURL, toURL, newBranchName, mes
 	reposToDelete = append(reposToDelete, destination)
 
 	var copied []string
+
 	if isLocal {
-		copied, err = local.CopyConfig(serviceName, localSource, destination)
-		if err != nil {
-			return fmt.Errorf("failed to set up local repository: %w", err)
+		if destination != nil {
+			overrideTargetFolder, err := destination.GetUniqueEnvironmentFolder()
+			if err != nil {
+				return fmt.Errorf("could not determine unique environment name for destination repository, error: %s", err.Error())
+			}
+			if overrideTargetFolder == "" {
+				return fmt.Errorf("could not determine destination environment name")
+			}
+			copied, err = local.CopyConfig(serviceName, localSource, destination, path.Join("environments", overrideTargetFolder))
+			if err != nil {
+				return fmt.Errorf("failed to set up local repository: %w", err)
+			}
 		}
 	} else {
-		copied, err = git.CopyService(serviceName, source, destination)
+		sourceEnvironment := ""
+		destinationEnvironment := ""
+		sourceEnvironment, err := source.GetUniqueEnvironmentFolder()
+		if err != nil {
+			return fmt.Errorf("could not determine unique environment name for source repository, error: %s", err.Error())
+		}
+
+		// Shouldn't hit this, but if so... we don't wanna continue as that'd go panic
+		if sourceEnvironment == "" {
+			return fmt.Errorf("could not determine source environment name")
+		}
+
+		destinationEnvironment, err = destination.GetUniqueEnvironmentFolder()
+
+		if destinationEnvironment == "" {
```

Comment:
I thought about that, so in that function itself there are lots of different errors that can occur and I think if we did, we'd lose the displaying of those - assuming you're thinking of replacing the 

`return "", err` calls with that string, for example

I'm wondering if I just drop this line and go with the err, because like with a previous comment, you can't have a directory called nothing anyway, and if something did go wrong you'd get an error